cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/connector/ConnectorSpec.java

```diff
@@ -17,20 +17,39 @@
 
 package io.cdap.cdap.etl.api.connector;
 
+import io.cdap.cdap.api.data.schema.Schema;
+
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
 
 /**
  * The connector spec contains all the properties based on the path and plugin config
  */
 public class ConnectorSpec {
+  // schema is null when the connector is unable to retrieve it from the resource
+  private final Schema schema;
+  private final Set<PluginSpec> relatedPlugins;
   private final Map<String, String> properties;
 
-  private ConnectorSpec(Map<String, String> properties) {
+  private ConnectorSpec(@Nullable Schema schema, Set<PluginSpec> relatedPlugins, Map<String, String> properties) {
+    this.schema = schema;
```

Comment:
so for specification endpoint, we need actually connect to the resource to get the schema ?

Reply:
Yes, this is to prevent UI to create a workspace and delete it when browsing from studio

Reply:
in this case, shall we allow Connector.generateSpec to throw IOException ?

Reply:
Added